src/Backend/Modules/Location/Engine/Model.php

```diff
@@ -149,142 +124,159 @@ public static function getCoordinates(
+                'extra_label' => SpoonFilter::ucfirst(Language::lbl('Location', 'Core'))
+                    . ': '
+                    . $location->getTitle(),
+                'language' => $location->getLocale()->getLocale(),
+                'edit_url' => BackendModel::createUrlForAction('Edit', 'Blog')
+                    . '&id='
+                    . $location->getId(),
             ]
         );
 
-        return $item['id'];
+        return $location->getId();
     }
 
-    /**
-     * Save the map settings
-     *
-     * @param int $mapId
-     * @param string $key
-     * @param mixed $value
-     */
-    public static function setMapSetting(int $mapId, string $key, $value): void
+    public static function setMapSetting(int $locationId, string $name, $value): void
     {
-        $value = serialize($value);
+        /** @var Location|null $location */
+        $location = self::getLocationRepository()->find($locationId);
+
+        if (!$location instanceof Location) {
+            throw new InvalidArgumentException('Location with id ' . $locationId . ' doesn\'t exist');
+        }
+
+        $setting = self::getLocationSettingRepository()->findOneBy(
+            [
+                'location' => $location,
+                'name' => $name
+            ]
+        );
+
+        if ($setting instanceof LocationSetting) {
+            $setting->update($value);
+
+            self::getEntityManager()->flush($setting);
+
+            return;
+        }
 
-        BackendModel::getContainer()->get('database')->execute(
-            'INSERT INTO location_settings(map_id, name, value)
-             VALUES(?, ?, ?)
-             ON DUPLICATE KEY UPDATE value = ?',
-            [$mapId, $key, $value, $value]
+        $location->addSetting(
+            new LocationSetting(
+                $location,
+                $name,
+                $value
+            )
         );
+
+        self::getEntityManager()->flush($location);
     }
 
-    /**
-     * Update an item
-     *
-     * @param array $item The data of the record to update.
-     *
-     * @return int
-     */
     public static function update(array $item): int
     {
-        // redefine edited on date
-        $item['edited_on'] = BackendModel::getUTCDate();
-
-        // we have an extra_id
-        if (isset($item['extra_id'])) {
-            // update extra
-            BackendModel::updateExtra(
-                $item['extra_id'],
-                'data',
-                [
-                    'id' => $item['id'],
-                    'extra_label' => \SpoonFilter::ucfirst(BL::lbl('Location', 'Core')) . ': ' . $item['title'],
-                    'language' => $item['language'],
-                    'edit_url' => BackendModel::createUrlForAction('Edit') . '&id=' . $item['id'],
-                ]
-            );
+        $currentLocation = self::getLocationRepository()->find($item['id']);
+
+        if (!$currentLocation instanceof Location) {
+            return 0;
```

Comment:
wouldn't it be better to throw an exception here

Reply:
The old implementation would return `0` if the `map_id` was non-existent.